clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -582,6 +582,13 @@ public int hashCode() {
         return originals.hashCode();
     }
 
+    /**
+     * @return true if the input map is a recording map. otherwise, false
+     */
+    public static boolean isRecording(Map<String, ?> map) {
```

Comment:
common/config/* is part of the public interface. This method seems internal. So, could we not expose it publicly to the end user?